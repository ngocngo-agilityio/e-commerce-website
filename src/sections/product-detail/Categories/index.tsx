// Libs
import { Flex, Text } from '@chakra-ui/react';

// Apis
import { getCategoryList } from '@apis';

interface Props {
  categoryIds: string[];
}

const Categories = async ({ categoryIds }: Props): Promise<JSX.Element> => {
  // Fetch data for categories
  const { data: categoryList = [] } = await getCategoryList({
    ids: categoryIds,
  });

  const categories = categoryList.map((item) => item.name).join(', ');

  return (
    <Flex>
      <Text
        as="span"
        size="md"
        lineHeight="24px"
        color="productDetail.categoryTitle"
      >
        Category:&nbsp;
      </Text>
      <Text size="md" lineHeight="24px">
        {categories}
      </Text>
    </Flex>
  );
};

export default Categories;
