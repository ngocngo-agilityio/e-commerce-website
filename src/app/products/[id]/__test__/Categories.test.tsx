// Libs
import { render, screen } from '@testing-library/react';

// Apis
import { getCategoryList } from '@apis';

// Mocks
import { MOCK_CATEGORIES } from '@mocks';

// Components
import Categories from '../Categories';

jest.mock('@apis');

(
  getCategoryList as jest.MockedFunction<typeof getCategoryList>
).mockResolvedValue({
  data: MOCK_CATEGORIES,
});

describe('Categories', () => {
  test('should render Categories successfully', async () => {
    const productDetailPage = await Categories({ categoryIds: ['1'] });

    render(productDetailPage);

    expect(screen.getByText('Category:')).toBeInTheDocument();
  });

  test('should match snapshot for Categories', async () => {
    const productDetailPage = await Categories({ categoryIds: ['1'] });

    const { container } = render(productDetailPage);

    expect(container).toMatchSnapshot();
  });
});
