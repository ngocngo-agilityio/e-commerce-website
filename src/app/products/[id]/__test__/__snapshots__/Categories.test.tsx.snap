// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories should match snapshot for Categories 1`] = `
<div>
  <div
    class="css-k008qs"
  >
    <span
      class="chakra-text css-1c9wxup"
    >
      Category:Â 
    </span>
    <p
      class="chakra-text css-1wsccn1"
    >
      Women, Men, Polo, Jacket, Casual, Formal
    </p>
  </div>
</div>
`;
